go-controller/pkg/ovn/ovn.go

```diff
@@ -754,6 +755,29 @@ func (oc *Controller) removeServiceEndpoints(lb, vip string) {
 	}
 }
 
+func (oc *Controller) assignPodAddresses(nodeName string) (net.HardwareAddr, *net.IPNet, error) {
+	var (
+		podMAC  net.HardwareAddr
+		podCIDR *net.IPNet
+		err     error
+	)
+
+	// First try to get the pod addresses quickly then fall back to polling every second.
+	podCIDR, err = oc.lsCache.AllocateNextIP(nodeName)
+	if err != nil {
+		return nil, nil, err
+	}
+	podMAC, err = hwaddr.GenerateHardwareAddr4(podCIDR.IP, hwaddr.PrivateMACPrefix)
+	if err != nil {
```

Comment:
L770: can you use our own `util.IPAddrToHWAddr`, that works for both ipv4 and ipv6

Reply:
ack